lightning/src/routing/network_graph.rs

```diff
@@ -428,6 +428,64 @@ impl Readable for NodeInfo {
 	}
 }
 
+/// Allows for updating user's network metadata.
+pub trait RouteFeePenalty {
+	/// Gets a channel's fee penalty based on its channel_id (as stored in a NetworkGraph object).
+	fn get_channel_fee_penalty(&self, chan_id: u64) -> Option<&u64>;
```

Comment:
I dont know that this needs the `Option<&` part - its easy to just return 0 if there is no penalty.